src/Features/CSharp/Portable/Completion/CompletionProviders/OperatorIndexerCompletionProvider.cs

```diff
@@ -0,0 +1,234 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Diagnostics;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Completion;
+using Microsoft.CodeAnalysis.Completion.Providers;
+using Microsoft.CodeAnalysis.CSharp.Extensions;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.LanguageServices;
+using Microsoft.CodeAnalysis.Options;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Text;
+using static Microsoft.CodeAnalysis.CSharp.SyntaxFactory;
+
+namespace Microsoft.CodeAnalysis.CSharp.Completion.Providers
+{
+    [ExportCompletionProvider(nameof(OperatorIndexerCompletionProvider), LanguageNames.CSharp), Shared]
+    [ExtensionOrder(After = nameof(SymbolCompletionProvider))]
+    internal class OperatorIndexerCompletionProvider : LSPCompletionProvider
+    {
+        [ImportingConstructor]
+        [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+        public OperatorIndexerCompletionProvider()
+        {
+        }
+
+        private const string MinimalTypeNamePropertyName = "MinimalTypeName";
+        private const string CompletionHandlerPropertyName = "CompletionHandler";
+        private const string CompletionHandlerConversion = "Conversion";
+        private const string CompletionHandlerIndexer = "Indexer";
+
+        // Identifier can start with "A Unicode character of classes Lu, Ll, Lt, Lm, Lo, or Nl". https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/language-specification/lexical-structure#identifiers
+        // Sorting is done via StringComparer.OrdinalIgnoreCase which compares the utf-16 bytes after converting to uppercase
+        // utf-16 \ufffd is the largest possible value for utf-16 and is also greater than surrogate pairs, if byte comparison is used.
+        // The "biggest" possible characters are \u3134a http://www.fileformat.info/info/unicode/char/3134a/index.htm surrogate pair "\ud884\udf4a" and
+        // \uffdc http://www.fileformat.info/info/unicode/char/ffdc/index.htm (non-surrogate)
+        private const string SortingPrefix = "\uFFFD";
```

Comment:
the comment oesn't reallyexplain what this is *for*.  is this basically just to ensure that all the conversions show up at the end?

also, who shows up more at the end, conversions, or "add import suggestions"?

Reply:
I added some more context to the code comment in e87d33a2. I can remove the pat with the "biggest" possible characters if you like.

> also, who shows up more at the end, conversions, or "add import suggestions"?

*Add import* comes after the indexer/conversions: 
![image](https://user-images.githubusercontent.com/24472128/94150119-095fe980-fe79-11ea-8726-b0ccea51ce2a.png)

I don't have tests for the "add import" case. Do we need such tests?

Reply:
having AddImport be last feels appropriate to me.  the first set are are the members thsi type directly exposes.  THe second is the set that can be indirectly brought in.

Reply:
Do we need a test, to make sure it does not regress?